corehq/apps/hqwebapp/management/commands/build_requirejs.py

```diff
@@ -213,7 +213,8 @@ def _get_main_js_modules_by_dir(html_files):
                 main = match.group(1)
                 directory = match.group(2)
                 if os.path.exists(os.path.join(ROOT_DIR, 'staticfiles', main + '.js')):
-                    dirs[directory].add(main)
+                    if not re.search(r'\bspec\b', main):
```

Comment:
This feels a little dangerous. I know all of our JS tests are in directories named `spec`, but could anything in production code ever match this. One obviously unlikely example is something like `some.spec.ial.file.js`

Maybe a slightly more targeted regex (matching only paths containing a directory named `spec`) would be worthwhile?
```suggestion
                    if not re.search(r'/spec/', main):
```

Reply:
yep, that's a good point

Reply:
Don't know why I didn't think of this previously, but ti does not need to use a regex. A simple string contains check would be must faster:
```py
if '/spec/' not in main:
```
Feel free to update that the next time you're working on that code.

Reply:
Added to the [requirejs for web apps branch](https://github.com/dimagi/commcare-hq/compare/jls/web-apps-requirejs), which is now only a couple of failing tests away from being PR-able 🤞